File: 携程8.23笔试/t1.cpp
#include <iostream>
#include <string>

int n;

int main() {
    std::cin >> n;
    int cur = 0, num = 0;
    std::string s;
    while (true) {
        s = std::to_string(num);
        if (cur + s.length() >= n) {
            std::cout << s[n - cur - 1];
            break;
        }
        else {
            cur += s.length();
            num += 2;
        }
    }
    return 0;
}

File: 携程8.23笔试/t1.py
import sys

def main():
    n = int(input().strip())
    cur = 0
    num = 0
    while True:
        s = str(num)
        if cur + len(s) >= n:
            print(s[n - cur - 1])
            break
        else:
            cur += len(s)
            num += 2

if __name__ == "__main__":
    main()